include/aspect/mesh_deformation/interface.h

```diff
@@ -34,6 +34,11 @@
 #include <deal.II/base/mg_level_object.h>
 #include <deal.II/lac/la_parallel_vector.h>
 #include <deal.II/multigrid/mg_transfer_matrix_free.h>
+#include <deal.II/matrix_free/operators.h>
```

Comment:
Same with all of these: move to the .cc. You should only need MGConstrainedDofs here.